scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -369,7 +369,8 @@ def generate_command(self):
             f'--junit-xml={self.report_file}',
             '--log-file-level=DEBUG',
             '--log-file-format=%(asctime)s.%(msecs)d:%(levelname)s:%(name)s: %(message)s',
-            f'--log-file={self.pytest_log_file_path}'
+            f'--log-file={self.pytest_log_file_path}',
+            f'--platform={self.instance.platform.name}'
```

Comment:
Hmmm... it seems `CMakeCache.txt` is not available for pytest-twister-harness own tests (see failing tests). Any ideas, @gchwier?

Reply:
because `CMakeCache,txt` is an output file from the build. When running unit tests, you do not have build artefacts (unless you provide some temp file or mock it).
Idea? Maybe it would be better to not modify any Twister/pytest-harness files, you can add a fixture just in your project (in test_edk.py or create conftest.py in pytest directory),
In that fixture you can also update a platform field:
```
import pytest
from twister_harness.helpers.utils import find_in_config

@pytest.fixture()
def unlaunched_dut(device_object: DeviceAdapter):
    """Return device object - with logs connected, but not run"""
    device_object.initialize_log_files('test_edk')
    try:
        if not device_object.device_config.platform:
            device_object.device_config.platform = find_in_config(
                device_object.device_config.build_dir / 'CMakeCache.txt', 'BOARD:STRING')
        yield device_object
    finally:
        device_object.close()
```

then you do not modify any common Twister files. If this fixture will be used in couple other tests, then it can be moved to existing fixtures, added to docs etc.

Reply:
Then I don't think this is any better than sending platform via command line args: if someone sees a test using  `device_config.platform` as is, and tries it on something else than a `unlaunched_dut`, they will be quite surprised to know it doesn't work. Guess I'll revert to the command line argument again, this way, things will be more consistent. Also, it seems clear it was meant to be this way, just an oversight not passing that argument.